agent/consul/state/config_entry.go

```diff
@@ -214,6 +214,15 @@ func (s *Store) ensureConfigEntryTxn(tx *memdb.Txn, idx uint64, conf structs.Con
 		return err // Err is already sufficiently decorated.
 	}
 
+	// If the config entry is for terminating gateways we update the memdb table
+	// that associates gateways <-> services.
+	if conf.GetKind() == structs.TerminatingGateway {
+		err = s.updateTerminatingGatewayServices(tx, idx, conf, entMeta)
```

Comment:
Do you have to do a similar thing to clean up indexed when a terminating gateway config entry is deleted?

Reply:
Missed that, thanks